packages/liveblocks-react/src/umbrella-store.ts

```diff
@@ -1299,17 +1306,24 @@ export class UmbrellaStore<M extends BaseMetadata> {
     query: ThreadsQuery<M> | undefined
   ) {
     const threadsFetcher = async (cursor?: string) => {
-      const room = this._client.getRoom(roomId);
-      if (room === null) {
-        throw new HttpError(`Room '${roomId}' is not available on client`, 479);
-      }
-
-      const result = await room.getThreads({ cursor, query });
+      const result = await this._client[kInternal].httpClient.getThreads<M>({
+        roomId,
+        cursor,
+        query,
+      });
       this.updateThreadsAndNotifications(
         result.threads,
         result.inboxNotifications
       );
 
+      this._store.set((state) => ({
+        ...state,
+        roomAccesses: {
+          ...state.roomAccesses,
+          [roomId]: result.roomAccess,
+        },
+      }));
```

Comment:
> only replacing the scopes if the new scope provides higher (write) permissions to the user?

I don't think we'll quickly run into issues with the current implementation, but I think it will be a bit better (more defensive) to implement this in a way that will only ever grow permissions. As if you're throwing all permissions you're ever seeing for a room into a set but never take anything away from there.

The reason is that in theory users can have multiple auth tokens in their auth manager that offer some access to the same room. For example, suppose users have Access Tokens A and B in their local cache:

1. Access token A offers `"engineering:room-a": ["room:write"]`
1. Access token B offers `"engineering:*": ["room:read"]`

Now, if you read threads for room `engineering:room-a`, then the permission hint for that room will be different depending on which access token you use to authorize that request.

This is why I think only ever adding permission hints to the existing set is best: if you have ever seen proof that you have write permission to a room: let's assume (and keep assuming) they have write access. If you see `["room:read"]` at a later time, we should definitely not overwrite the hint. We may still have write access to that room!

It's the job of the auth manager to give you the most permissive token in the cache for a given outgoing request anyway, so we should be able to assume that if the user is creating a new thread in room `engineering:room-a`, they will do so with token A (as provided by the auth manager), whereas if creating a new thread for room `engineering:room-bar` they will be handed token B to do so.

Reply:
Thank you for the thorough explanation 🙏🏼 Definitely a tricky one to get absolutely right. I have updated the logic to only ever add permission to the permission set. So, if a `room:write` permission has been recorded for a room and a new `room:read` permission comes along, we do not replace the permission set with the single `room:read` permission but rather add the `room:read` permission too, so that the new permission set for the room becomes `["room:write", "room:read"]`. It felt simpler to do this. 

```ts
private updateRoomPermissions(permissions: Record<string, Permission[]>) {
  const permissionsByRoom = this._store.get().permissionsByRoom;

  Object.entries(permissions).forEach(([roomId, permissions]) => {
    // Get the existing set of permissions for the room and only ever add permission to this set
    const existingPermissions = permissionsByRoom[roomId];

    if (existingPermissions === undefined) {
      permissionsByRoom[roomId] = new Set(permissions);
    } else {
      permissions.forEach(existingPermissions.add);
    }
  });

  this._store.set((state) => ({
    ...state,
    permissionsByRoom,
  }));
}
```

Reply:
Yes, perfect! 👍